app/SqsQueueConsumer.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import javax.inject.{Inject, Provider}
+
+import com.amazonaws.services.sqs.model.{ReceiveMessageRequest, ReceiveMessageResult}
+import com.amazonaws.services.sqs.{AmazonSQS, AmazonSQSClientBuilder}
+import model.Message
+import play.api.Configuration
+import uk.gov.hmrc.play.config.ServicesConfig
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+import com.amazonaws.services.sqs.model.DeleteMessageRequest
+
+class SqsQueueConsumer @Inject()(sqsClient: AmazonSQS, configuration: Configuration, environment: play.api.Environment)(
```

Comment:
What do you think about creating separate, type-safe configuration object which you inject.
So that you don't have to inject configuration and environment? See upscan-initiate.